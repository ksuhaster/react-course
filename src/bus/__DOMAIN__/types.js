export const types = Object.freeze({
    DOMAIN_ACTION: 'DOMAIN_ACTION'
});