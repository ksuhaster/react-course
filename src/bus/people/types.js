export const types = Object.freeze({
    // Sync
    PEOPLE_START_FETCHING: 'PEOPLE_START_FETCHING',
    PEOPLE_STOP_FETCHING: 'PEOPLE_STOP_FETCHING',
    PEOPLE_FILL: 'PEOPLE_FILL',

    // Async
    PEOPLE_FETCH_ASYNC: 'PEOPLE_FETCH_ASYNC',
});
