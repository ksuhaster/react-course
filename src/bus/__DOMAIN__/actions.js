// Types
import { types } from './types';

export const domainActions = Object.freeze({
    domainAction: () => {
        return {
            type: types.DOMAIN_ACTION
        }
    }
});