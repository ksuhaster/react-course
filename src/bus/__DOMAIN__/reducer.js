// Types
import { types } from './types';

const initialState = {
    property: ''
};

export const domainReducer = ( state = initialState, { type } ) => {
    switch (type) {
        case types.DOMAIN_ACTION:
            return { ...state, property: '' };
    
        default:
            return state;
    }
}