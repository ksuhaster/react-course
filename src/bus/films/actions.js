// Types
import { types } from './types';

import { api } from '../../api';

export const filmsActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILMS_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILMS_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILMS_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILMS_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.FILMS_FETCH_ASYNC
        });

        dispatch(filmsActions.startFetching());

        const response = await api.films.fetch();

        if (response.status === 200) {
            const { results } = await response.json();
            dispatch(filmsActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(filmsActions.setFetchingError(error));
        }

        dispatch(filmsActions.stopFetching());
    }
});


export const filmActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILM_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILM_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILM_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILM_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    fetchAsync: (id) => async (dispatch) => {
        dispatch({
            type: types.FILM_FETCH_ASYNC
        });

        dispatch(filmActions.startFetching());

        const response = await api.film.fetch(id);

        if (response.status === 200) {
            const results = await response.json();
            dispatch(filmActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(filmActions.setFetchingError(error));
        }

        dispatch(filmActions.stopFetching());
    }
});