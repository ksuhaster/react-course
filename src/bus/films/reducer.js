// Types
import { types } from './types';

const initialState = {
    data: [],
    isFetching: false,
    error: false
};

export const filmsReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.FILMS_START_FETCHING:
            return { ...state, isFetching: true };
        case types.FILMS_STOP_FETCHING:
            return { ...state, isFetching: false };
        case types.FILMS_SET_FETCHING_ERROR:
            return { ...state, error: payload };
        case types.FILMS_FILL:
            return { ...state, data: payload };

        default:
            return state;
    }
};

export const filmReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.FILM_START_FETCHING:
            return { ...state, isFetching: true };
        case types.FILM_STOP_FETCHING:
            return { ...state, isFetching: false };
        case types.FILM_SET_FETCHING_ERROR:
            return { ...state, error: payload };
        case types.FILM_FILL:
            return { ...state, data: payload };

        default:
            return state;
    }
};