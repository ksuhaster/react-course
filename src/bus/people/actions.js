// Types
import { types } from './types';

import { api } from '../../api';

export const peopleActions = Object.freeze({
    // Sync
    fill: (payload) => {
        return {
            type: types.PEOPLE_FILL,
            payload
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.PEOPLE_FETCH_ASYNC
        });

        const response = await api.people.fetch();
        const { results } = await response.json();

        dispatch(peopleActions.fill(results));
    }
});