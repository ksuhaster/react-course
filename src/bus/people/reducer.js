// Types
import { types } from './types';

const initialState = {
    data: [],
    isFetching: false,
    error: false
};

export const peopleReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.PEOPLE_START_FETCHING:
            return { ...state, isFetching: true };
        case types.PEOPLE_STOP_FETCHING:
            return { ...state, isFetching: false };
        case types.PEOPLE_SET_FETCHING_ERROR:
            return { ...state, error: payload };
        case types.PEOPLE_FILL:
            return { ...state, data: payload };

        default:
            return state;
    }
};