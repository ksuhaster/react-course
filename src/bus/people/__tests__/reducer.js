import { peopleReducer, personReducer } from '../reducer';
import { peopleActions, personActions } from '../actions';
import { types } from "../types";


describe('person list reducer:', () => {
    test('should return initial state by default', () => {
        expect(peopleReducer(void 0, {})).toMatchSnapshot();
    });

    // Sync
    test(`should handle ${types.PEOPLE_START_FETCHING} action`, () => {
        expect(peopleReducer(void 0, peopleActions.startFetching())).toMatchSnapshot();
    });
    test(`should handle ${types.PEOPLE_STOP_FETCHING} action`, () => {
        expect(peopleReducer(void 0, peopleActions.stopFetching())).toMatchSnapshot();
    });
    test(`should handle ${types.PEOPLE_SET_FETCHING_ERROR} action`, () => {
        expect(peopleReducer(void 0, peopleActions.setFetchingError())).toMatchSnapshot();
    });
    test(`should handle ${types.PEOPLE_FILL} action`, () => {
        expect(peopleReducer(void 0, peopleActions.fill())).toMatchSnapshot();
    });

    // Async
    test(`should handle ${types.PEOPLE_FETCH_ASYNC} action`, () => {
        expect(peopleReducer(void 0, peopleActions.fetchAsync())).toMatchSnapshot();
    });
});


describe('one person reducer:', () => {
    test('should return initial state by default', () => {
        expect(personReducer(void 0, {})).toMatchSnapshot();
    });

    // Sync
    test(`should handle ${types.PERSON_START_FETCHING} action`, () => {
        expect(personReducer(void 0, personActions.startFetching())).toMatchSnapshot();
    });
    test(`should handle ${types.PERSON_STOP_FETCHING} action`, () => {
        expect(personReducer(void 0, personActions.stopFetching())).toMatchSnapshot();
    });
    test(`should handle ${types.PERSON_SET_FETCHING_ERROR} action`, () => {
        expect(personReducer(void 0, personActions.setFetchingError())).toMatchSnapshot();
    });
    test(`should handle ${types.PERSON_FILL} action`, () => {
        expect(personReducer(void 0, personActions.fill())).toMatchSnapshot();
    });

    // Async
    test(`should handle ${types.PERSON_FETCH_ASYNC} action`, () => {
        expect(personReducer(void 0, personActions.fetchAsync())).toMatchSnapshot();
    });
});
