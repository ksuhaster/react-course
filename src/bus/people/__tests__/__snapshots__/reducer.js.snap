// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`one person reducer: should handle PERSON_FETCH_ASYNC action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;

exports[`one person reducer: should handle PERSON_FILL action 1`] = `
Object {
  "data": undefined,
  "error": false,
  "isFetching": false,
}
`;

exports[`one person reducer: should handle PERSON_SET_FETCHING_ERROR action 1`] = `
Object {
  "data": Array [],
  "error": undefined,
  "isFetching": false,
}
`;

exports[`one person reducer: should handle PERSON_START_FETCHING action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": true,
}
`;

exports[`one person reducer: should handle PERSON_STOP_FETCHING action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;

exports[`one person reducer: should return initial state by default 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;

exports[`person list reducer: should handle PEOPLE_FETCH_ASYNC action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;

exports[`person list reducer: should handle PEOPLE_FILL action 1`] = `
Object {
  "data": undefined,
  "error": false,
  "isFetching": false,
}
`;

exports[`person list reducer: should handle PEOPLE_SET_FETCHING_ERROR action 1`] = `
Object {
  "data": Array [],
  "error": undefined,
  "isFetching": false,
}
`;

exports[`person list reducer: should handle PEOPLE_START_FETCHING action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": true,
}
`;

exports[`person list reducer: should handle PEOPLE_STOP_FETCHING action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;

exports[`person list reducer: should return initial state by default 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;
